### DISCUSSION TITLE
ulimit -nの変更について

### DISCUSSION THREAD

**Discussion by motthy** [nig-sc/nigsc_homepage2#452] [Q&A] [answered]
ulimit -n の設定が1024だったので、もっと大きな値に変更するため、
`ulimit -n 131072`
を`.bash_profile`に追記しました。
その後、qsubでsungularityコンテナを利用したスクリプトを実行すると、stderrに
```
/home/usename/.bash_profile: line 26: ulimit: open files: cannot modify limit: Operation not permitted
```
というメッセージが出ます。
ulimit -nの値を変更すると不都合なのでしょうか？
できればデフォルトの値（1024）ではなく、十分大きな値（>100000）を初期設定としていただくことは可能でしょうか？

ご参考まで、実行したスクリプトは以下です
```
#$ -S /bin/bash
#$ -cwd
#$ -pe def_slot 8
#$ -l mem_req=15G,s_vmem=15G
#$ -l d_rt=720:00:00
#$ -l s_rt=720:00:00

SIMS="singularity exec /usr/local/biotools/l/liftoff:1.6.1--pyhdfd78af_0"
$SIMS liftoff -g genome,gff -p 8 -o liftoff.gff -u unmapped_features.txt \
                       -copies -dir tmp target_genome.fa refgenome.fa
```

**Comment by sc-fs** [comment, marked as answer]
qsub時に発生するエラーの原因ですが、
こちらは、AGE（ジョブスケジュール）側で設定している制限値：8192
以上の値を設定（ulimit -n 131072）されたのがエラーの原因です。

現在、AGE側の制限値をAGE起動中に変更する方法を確認しております。

お手数をお掛けしますが、変更が出来るまで、
現時点での最大値8192にて稼働していただきますようお願いします。

**Reply by motthy** [reply]
回答ありがとうございました。変更していただけるまで`ulimit -n 8192`に設定して利用します。

> できればデフォルトの値（1024）ではなく、十分大きな値（>100000）を初期設定としていただくことは可能でしょうか？

上記もご検討いただけると助かります。

**Reply by sc-fs** [reply]
連絡が遅れ申し訳ありません。ulimit -n 131072（最大値）を指定可能になりました。
